content/posts/2022-01-29-things-i-learned-in-2021.md

```diff
@@ -0,0 +1,116 @@
+
+![Dunning-Kruger Effect](/media/2021/dunningkruger.png)
+
+## Updated my open source library to version 2.0.0 and got ~4000 downloads per week
+
+![linked in package](/media/2021/package.png)
+
+I did release an [open source package](https://www.npmjs.com/package/react-linkedin-login-oauth2) 4 years ago (2018). It basically helps you to integrate the Login with Linkedin feature to a React application easily. When I start this, I just do it for fun and thought no one was gonna use this library. But after a long time, I see that there are some people who actually installed my package. I also received bug reports and feature requests. (All of that happens without zero advertising). I decided to spend some weekends working on this little project. People started to thank me for helping them. I felt wonderful that my work can save some people’s time. I also learned a lot (I will write a post on what I learned when doing open source). I realized that I actually received more than I gave to the community. In October, I wanted to [release the package to version 2.x](https://github.com/nvh95/react-linkedin-login-oauth2/pull/50) to tackle some major problems in version 1.x. I learned a lot. I converted my code to Typescript, I learned more about Typescript. I used Rollup to bundle my code instead of webpack to support [ECMAScript modules](https://nodejs.org/api/esm.html), to leverage the [Tree Shaking](https://developer.mozilla.org/en-US/docs/Glossary/Tree_shaking) feature. For the first time, I knew about Vite, and use it for the example while developing the package. I had so much fun working on these items and learn so much. After investing some time in the project, I see the strike in the number of weekly downloads that amazed me. It reached about 4000 downloads per week. There are many things to do for this little project and I hope I can make it better in 2022.
+
+## Participated more actively in the open source world
+
+![opensource](/media/2021/opensource.png)
+
+I love open source. I use open source software every-single-day. I use React, Node.js, VS Code and many many other things. Humanity can go that far I think open source plays a big role. The good thing about open source is that everyone can make their lives and others’ lives much easier and more enjoyable. Anyone can contribute to the open source. It does not require you have to write code to core services of a framework or application. It can be anything, from a typo to the recommendations, the feedback about the software. Next year, I target to contribute more to some projects that I like and am interested like msw and Next.js.
```

Comment:
`Humanity can go that far I think open source plays a big role.` --> I think open source plays a big role in bringing humanity this far.

Reply:
This paragraph is a little bit off when reading the text only. You can change the beginning to `Anyone can contribute to the open source, and so did I`. And then, provide your reasons more after that.